### PR TITLE
backupccl: deflake multinode datadriven tests that set cluster settings

### LINKED ISSUE #92808: ccl/backupccl: TestDataDriven failed
ccl/backupccl.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7783970?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7783970?buildTab=artifacts#/) on release-22.2 @ [ca673993473629a46370100108dd8c185a592e91](https://github.com/cockroachdb/cockroach/commits/ca673993473629a46370100108dd8c185a592e91):

```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2921/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/multiregion:79:
        exec-sql [0 args]
        BACKUP DATABASE no_region_db INTO 'nodelocal://1/no_region_database_backup/';
        ----
    datadriven_test.go:383: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2921/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/multiregion:83:
        exec-sql [0 args]
        BACKUP INTO 'nodelocal://1/no_region_cluster_backup/';
        ----
    datadriven_test.go:383: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2921/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/multiregion:87:
        exec-sql [0 args]
        DROP DATABASE no_region_db;
        ----
    datadriven_test.go:383: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2921/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/multiregion:91:
        exec-sql [0 args]
        DROP DATABASE no_region_db_2;
        ----
    datadriven_test.go:383: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2921/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/multiregion:95:
        exec-sql [1 args]
        SET CLUSTER SETTING sql.defaults.primary_region = 'non-existent-region';
        ----
    datadriven_test.go:383: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2921/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/multiregion:99:
        exec-sql [0 args]
        RESTORE DATABASE no_region_db FROM LATEST IN 'nodelocal://1/no_region_database_backup/';
        ----
        pq: region "non-existent-region" does not exist
        HINT: valid regions: eu-central-1, eu-north-1
        --
        set the default PRIMARY REGION to a region that exists (see SHOW REGIONS FROM CLUSTER) then using SET CLUSTER SETTING sql.defaults.primary_region = 'region'
    datadriven_test.go:383: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2921/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/multiregion:107:
        exec-sql [1 args]
        SET CLUSTER SETTING sql.defaults.primary_region = 'eu-central-1';
        ----
    datadriven_test.go:383: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2921/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/multiregion:111: RESTORE DATABASE no_region_db FROM LATEST IN 'nodelocal://1/no_region_database_backup/';
        expected:
        NOTICE: setting the PRIMARY REGION as eu-central-1 on database no_region_db
        HINT: to change the default primary region, use SET CLUSTER SETTING sql.defaults.primary_region = 'region' or use RESET CLUSTER SETTING sql.defaults.primary_region to disable this behavior
        
        found:
        pq: region "non-existent-region" does not exist
        HINT: valid regions: eu-central-1, eu-north-1
        --
        set the default PRIMARY REGION to a region that exists (see SHOW REGIONS FROM CLUSTER) then using SET CLUSTER SETTING sql.defaults.primary_region = 'region'
    --- FAIL: TestDataDriven/multiregion (32.52s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92646 ccl/backupccl: TestDataDriven failed [C-test-failure O-robot T-disaster-recovery branch-master]
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-21979

### PR DESCRIPTION
**Pull request by msbutler**
The `SET CLUSTER SETTING` stmt propogrates the cluster setting to remote nodes asynchronously, but our data driven multinode tests incorrectly assumed that settings were propogated during stmt execution. This patch adds a new 'set-cluster-setting' dd cmd which will propogate the cluster setting to all nodes before the cmd returns.

Fixes #92808

Release note: none

### CODE REVIEW

File: pkg/ccl/backupccl/datadriven_test.go [hunk 1]
```diff
@@ -87,6 +87,10 @@ type sqlDBKey struct {
 }
 
 type datadrivenTestState struct {
+	// clusters maps a name to its cluster
+	clusters map[string]serverutils.TestClusterInterface
```
Comment (msbutler) [inline pkg/ccl/backupccl/datadriven_test.go:91]: I wonder if we ought to change the `new-server` cmd to `new-cluster`, as it's really spinning up a new cluster and pushing all sql cmds to get served by the first server in the test cluster.

**Comment by msbutler** [comment]
nice sleuthing! for posterity, it looks like this bug may have been introduced by https://github.com/cockroachdb/cockroach/pull/89630

**Review by adityamaru — APPROVED** [review]

File: pkg/ccl/backupccl/datadriven_test.go (hunk 1 above)
Comment (adityamaru) [reply pkg/ccl/backupccl/datadriven_test.go:91]: sgtm 👍

Comment (msbutler) [reply pkg/ccl/backupccl/datadriven_test.go:91]: just pushed an extra commit for this.

**Comment by msbutler** [comment]
TFTR!

bors r=adityamaru

**Pull request merged**